lib/S3Plugin.js

```diff
@@ -361,6 +412,133 @@ class S3Plugin {
     return stringParam;
   }
 
+  /**
+   * validate given string to match
+   * AWS S3 bucket name format prerequisites
+   */
+    _validateBucketName (name, disableDotsInName=false) {
+      const bucketNameRegex = disableDotsInName ? 
+        // avoiding dots is needed for S3 Transfer Acceleration
+        /(?!(^xn--|.+-s3alias$))^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$/
+        : /(?!(^((2(5[0-5]|[0-4][0-9])|[01]?[0-9]{1,2})\.){3}(2(5[0-5]|[0-4][0-9])|[01]?[0-9]{1,2})$|^xn--|.+-s3alias$))^[a-z0-9][a-z0-9.-]{1,61}[a-z0-9]$/
+      
+        const sanitizedName = bucketNameRegex.test(name)
+  
+      if (sanitizedName === null) {
+        throw new this.context.errors.BadRequestError(`Invalid bucket name format: ${name}`)
+      }
+      
+      return true
+    }
+
+  /**
+   * Creates a new S3 bucket.
+   *
+   * @controller            bucket
+   * @action                create
+   *
+   * @param {KuzzleRequest} request
+   */
+  async bucketCreate (request) {
+    this._assertCredentials();
+
+    const
+      bucketName = this._stringArg(request, 'bucketName', this.config.bucketName),
+      bucketRegion = this._stringArg(request, 'bucketRegion', this.config.region),
+      disableDotsInName = request.input.body.disableDotsInName,
+      bucketPolicy = request.input.body.bucketPolicy,
+      defaultCORS = {
+        CORSRules : [ 
+          {
+            AllowedHeaders: ['*'],
+            AllowedMethods: ['GET', 'POST', 'PUT'],
+            AllowedOrigins: ['*'],
+          }
+        ]
+      },
+      bucketCORS = request.input.body.cors || defaultCORS;
+      
+    let bucketOptions = request.input.body.options;
+    
+    if (!bucketPolicy && !bucketOptions){
+      bucketOptions = {ACL : 'public-read'}
+    }
+
+    if (await this.bucketExists(request)){
+      // TODO: add 409 conflict error template to kuzzle ?
+      throw new this.context.errors.BadRequestError(`Bucket name already exist`);
+    }
+    if (!this._validateBucketName(bucketName, disableDotsInName)){
+      throw new this.context.errors.BadRequestError(`Bucket name ${bucketName} contains unauthorized character(s)`);
+    }
+    
+    AWS.config.update({ region: bucketRegion });
+
+    const params = { Bucket: bucketName };
+    
+    await this.s3.createBucket({ ...params, ...bucketOptions }).promise();
```

Comment:
It's probably not a problem but here you allow the user to overwrite the `Bucket` key if they pass it in the `bucketOptions` arg

Reply:
yes it is intended to allow user use aws-sdk s3 options